code/datums/status_effects/debuffs/cursed.dm

```diff
@@ -0,0 +1,187 @@
+			messages += span_userdanger("The toll has already been exacted. There is no longer death on 'your' terms. Is your dignity worth more than your life?")
+
+		if(5 to INFINITY)
+			if(max_curse_count != DEFAULT_MAX_CURSE_COUNT) // this probably will only happen through admin schenanigans letting people stack up infinite curses or something
+				to_chat(owner, span_boldwarning("Do you <i>still</i> think you're in control?"))
+				return
+
+			to_chat(owner, span_userdanger("Why couldn't I get one more try?!"))
+			owner.investigate_log("has been gibbed by the cursed status effect after accumulating [curse_count] curses.", INVESTIGATE_DEATHS)
+			owner.gib()
+			qdel(src)
+			return
+
+	for(var/message in messages)
+		to_chat(owner, message)
+		sleep(1.5 SECONDS) // yes yes a bit fast but it can be a lot of text and i want the whole thing to send before the cooldown on the slot machine might expire
+
+/// Cleans ourselves up and removes our curses. Meant to be done in a "positive" way, when the curse is broken. Directly use qdel otherwise.
+/datum/status_effect/grouped/cursed/proc/clear_curses()
+	SIGNAL_HANDLER
+
+	if(!isnull(branded_hand))
+		var/datum/wound/brand = branded_hand.get_wound_type(/datum/wound/burn/severe/cursed_brand)
+		if(!isnull(brand))
+			brand.remove_wound()
+
+	owner.visible_message(
+		span_notice("The smoke slowly clears from [owner.name]..."),
+		span_notice("Your skin finally settles down and your throat no longer feels as dry... The brand disappearing confirms that the curse has been lifted."),
+	)
+	QDEL_NULL(particle_effect)
+	qdel(src)
+
+/// If our owner's stat changes, rapidly surge the damage chance.
+/datum/status_effect/grouped/cursed/proc/on_stat_changed()
+	if(owner.stat == CONSCIOUS || owner.stat == DEAD) // reset on these two states
+		damage_chance = initial(damage_chance)
+		return
+
+	to_chat(owner, span_userdanger("As your body crumbles, you feel the curse of the slot machine surge through your body!"))
+	damage_chance += 75 //ruh roh raggy
+
+/// If our owner dies without getting gibbed (as in of other causes), stop smoking because we've "expended all the life energy".
+/datum/status_effect/grouped/cursed/proc/on_death(mob/living/source, gibbed)
+	SIGNAL_HANDLER
+
+	if(gibbed)
+		return
+
+	QDEL_NULL(particle_effect)
+
+/datum/status_effect/grouped/cursed/update_particles()
+	var/particle_path = /particles/smoke/steam/mild
+	switch(curse_count)
+		if(2 to 3)
+			particle_path = /particles/smoke/steam
+		if(4)
+			particle_path = /particles/smoke/steam/bad
+
+	if(smoke_path == particle_path)
+		return
+
+	QDEL_NULL(particle_effect)
+	smoke_path = particle_path
+	particle_effect = new(owner, particle_path)
+
+/datum/status_effect/grouped/cursed/tick(seconds_between_ticks)
+	if(curse_count <= 1)
+		return // you get one "freebie" (single damage) to nudge you into thinking this is a bad idea before the house begins to win.
+
+	// the house won.
+	var/ticked_coefficient = rand(0.15, 0.40)
```

Comment:
Rand does not work with decimals

Reply:
dammit i knew there was something weird with this shit